### ISSUE TITLE
get-token should probably return strings, not ints

### ISSUE THREAD

**Issue by boxxxie** [suvash/one-time#13]
`get-token` doesn't always return codes that look like what the user is supposed to type in, sometimes i get 5 or 4 digit codes

`(get-token "OV2K2K5SOWKN2RCZ" 1)  ;; => 3799`

so, I have to do some work to present this to the user (authenticator apps will pad with leading 0s)

also i have to do work to take what my form gives me, and turn that into an int.

there is possibly another problem, clojure does treat numbers with leading 0s differently, i haven't run into this being an issue, but i imagine it could be for some people/situations.

```
(str 00003123) ;; => "1619"
(str 3123) ;; => "3123"
```

I think it would be simpler to keep these tokens as strings. below is a proposed change to the get-token function
only the last line has been altered.

```
(defn get-token
  "Return a HOTP token (HMAC-Based One-Time Password Algorithm)
   based on a secret and a counter, as specified in
   https://tools.ietf.org/html/rfc4226"
  ([secret counter]
   ;; Use HMAC-SHA-1 as default when not provided
   (get-token secret counter :hmac-sha-1))
  ([secret counter hmac-sha-type]
   (let [digest (hmac-sha-digest secret counter hmac-sha-type)
         offset (bit-and (digest 19) 0xf)
         code (bit-or (bit-shift-left (bit-and (digest offset) 0x7f) 24)
                (bit-shift-left (bit-and (digest (+ offset 1)) 0xff) 16)
                (bit-shift-left (bit-and (digest (+ offset 2)) 0xff) 8)
                (bit-and (digest (+ offset 3)) 0xff))]
     (->>
       (rem code 1000000)
       (format "%06d")))))
```

**Comment by suvash** [comment]
Regarding the string vs. integer, I'd like to direct you to the specification (url included in the function description), section 5.3 if you're really in a hurry. https://datatracker.ietf.org/doc/html/rfc4226#section-5.3

<img width="627" alt="Screenshot 2021-07-11 at 01 18 28" src="https://user-images.githubusercontent.com/144952/125178543-60effb80-e1e6-11eb-877c-031aebf5bcda.png">


On the following concerns you've raised 
> so, I have to do some work to present this to the user (authenticator apps will pad with leading 0s)
> 
> also i have to do work to take what my form gives me, and turn that into an int.

YES, your application is going to be responsible for the presentation and parsing of the tokens. This library tries it's best to not have strict opinions on how application layer decisions should be made.

I'm hoping that clarifies. I'll close this issue for now, but please feel free to continue the conversation.
Cheers.

**Comment by boxxxie** [comment]
thanks for the reply, i guess the user interface for these codes made me think this lib was missing something, but instead its the interfaces that have added things.
